### ISSUE TITLE
Date Filter not working in js-grid using moment.js "DD/MM/YYYY" format. It works well with "MM/DD/YYYY"

### ISSUE THREAD

**Issue by RoshniUpadhyaya** [tabalinas/jsgrid#565]
I have problem filtering date using moment.js "DD/MM/YYYY" format. Filter applies only on the Day. Month & Year are not getting filter. Please suggest me the solution. It works well with "MM/DD/YYYY"

Below is the js-grid code

```js
function CallDCJSGrid(params) {
    var dcdate;
    var filterFlag;
    var DateField = function (config) {
        jsGrid.Field.call(this, config);
    };
    params.poId = params.poId || 0;
    //Below code is to check poId & set filtering to true if poId is found 0 otherwise set filtering to false
    if (params.poId == 0) {
        filterFlag = true;
    }
    else {
        filterFlag = false;
    }
    //Customization of Date field 
    DateField.prototype = new jsGrid.Field({
        itemTemplate: function (value) {
            if (value != null) {
                var dcdateFormat = moment(value).format("DD/MM/YYYY");
                return dcdateFormat;
            }
        },
        filterTemplate: function () {
            var now = new Date();
            this._fromPicker = $("<input id='frmDate' placeholder='From Date'>").datepicker({ dateFormat: "dd/mm/yy", maxDate: new Date(), defaultDate: now.setFullYear(now.getFullYear() - 1) });
            this._toPicker = $("<input id='toDate' placeholder='To Date'>").datepicker({ dateFormat: "dd/mm/yy", maxDate: new Date(), defaultDate: now.setFullYear(now.getFullYear() + 1) });
            return $("<div>").append(this._fromPicker).append(this._toPicker);
        },
        filterValue: function () {
            var fromDate = $('#frmDate').val();
            var toDate = $('#toDate').val();
            return {
                from: fromDate,
                to: toDate
            };
        }
    });
    //Assigning the custom datefields to grid date column
    jsGrid.fields.date = DateField;
    //Loading the grid for with all delivery challans.
    $("#DCJSGrid").jsGrid({
        width: "100%",
        height: "auto",
        sorting: true,
        filtering: filterFlag,
        paging: true,
        autoload: true,
        pageSize: params.pageSize,
        pageButtonCount: params.pageButtonCount,
        pagerContainer: "#DCJSGridPager",
        pagerFormat: "No of Records: {itemCount} | Pages: {first} {prev} {pages} {next} {last}    {pageIndex} of {pageCount}",
        pagePrevText: "Prev",
        pageNextText: "Next",
        pageFirstText: "First",
        pageLastText: "Last",
        pageNavigatorNextText: "...",
        pageNavigatorPrevText: "...",
        align: "center",
        controller: {
            loadData: function (filter) {
                // Below code is used to decide the data parameter of Ajax to get the DC list based on poId value
                var filterData = "";
                if (params.poId > 0) {
                    filterData = {
                        "poId": params.poId
                    };
                } else {
                    filterData = filter;
                }
                var d = $.Deferred();
                $.ajax({
                    url: params.GetDCDataUrl,
                    type: "GET",
                    contentType: "application/json",
                    data: filterData,
                    dataType: "json"
                }).done(function (response) {
                    response = $.grep(response, function (item) {
                        dcdate = moment(item.DCDate).format("DD/MM/YYYY");
                        if (filterFlag) {
                            // Applying filter
                            return (!filter.pPONumber || (item.pPONumber == null ? "" : item.pPONumber).toLowerCase().indexOf(filter.pPONumber.toLowerCase()) > -1)
                                && (!filter.DCNO || (item.DCNO == null ? "" : item.DCNO).toLowerCase().indexOf(filter.DCNO.toLowerCase()) > -1)
                                && (!filter.VendorCode || (item.VendorCode == null ? "" : item.VendorCode).toLowerCase().indexOf(filter.VendorCode.toLowerCase()) > -1)
                                && (!filter.pVendorName || (item.VendorCode == null ? "" : item.pVendorName).toLowerCase().indexOf(filter.pVendorName.toLowerCase()) > -1)
                                && (!filter.DCDate.from || dcdate >= filter.DCDate.from)
                                && (!filter.DCDate.to || dcdate <= filter.DCDate.to)
                                && (!filter.UploadDocument || item.UploadDocument.indexOf(filter.UploadDocument) > -1)
                        }
                        else {
                            return (!filter.pPONumber || (item.pPONumber == null ? "" : item.pPONumber).toLowerCase().indexOf(filter.pPONumber.toLowerCase()) > -1)
                                && (!filter.DCNO || (item.DCNO == null ? "" : item.DCNO).toLowerCase().indexOf(filter.DCNO.toLowerCase()) > -1)
                                && (!filter.VendorCode || (item.VendorCode == null ? "" : item.VendorCode).toLowerCase().indexOf(filter.VendorCode.toLowerCase()) > -1)
                                && (!filter.pVendorName || (item.VendorCode == null ? "" : item.pVendorName).toLowerCase().indexOf(filter.pVendorName.toLowerCase()) > -1)
                                && (!filter.UploadDocument || item.UploadDocument.indexOf(filter.UploadDocument) > -1)
                        }
                    });
                    d.resolve(response);
                }).fail(function (jqXHR, textStatus) {
                    var err = "Error " + " " + textStatus;
                    if (jqXHR.responseText && jqXHR.responseText[0] == "{")
                        err = JSON.parse(jqXHR.responseText).Message;
                    console.log(err);
                });
                return d.promise();
            }
        },
        fields: [
        { name: "DCNO", type: "text", width: 10, title: "D.C. Number", align: "center" },
        { name: "DCDate", type: "date", width: 19, align: "center" },
        { name: "pPONumber", type: "text", width: 19, align: "center", title: "P.O. Number" },
        { name: "VendorCode", type: "text", width: 17, title: "Vendor Code", align: "center", visible: false },
        { name: "pVendorName", type: "text", width: 45, title: "Vendor Name", visible: false },
        {
            name: "UploadDocument", type: "text", width: 4, sorting: false, title: "Document", align: "center",
            filtering: false, itemTemplate: function (value, item) {
                if (item.UploadDocument != null && item.UploadDocument != "") {
                    return $('<a>').attr("href", item.UploadDocument).attr("download", "download").attr("alt", "Download").attr("title", "Download").append('<img src="/Content/img/download.png" class="je-grid-icons" alt="document" />').append('</a>');
                }
            }
        },
        {
            type: "control", width: 2, modeSwitchButton: false, editButton: false, deleteButton: false,
            headerTemplate: function () {
                return $("<img>").attr("src", "/Content/Img/Add.png").attr("id", "imgCreateDC").attr("alt", "Create DC").attr("title", "Create DC").css({ "cursor": "pointer" })
                    .on("click", function () {
                        window.location.href = params.CreateDCUrl.replace("POID", params.poId);
                    });
            }
        }]
    });
}
```

**Comment by tabalinas** [comment]
If the issue is still actual, please, fork the following fiddle https://jsfiddle.net/tabalinas/v6ww51ez/
and modify it to demonstrate the problem.

**Comment by theeedannyd** [comment]
Hi,

I have been trying to isolate a problem for a while in two projects and I think mine is related to this. I have a custom date field that uses the custom field code supplied in the website but modified to use moment instead of date. Interestingly, sorting by that column always only appears to sort the first two rows. There may be something to do with the interaction of moment.js and jsgrid. I am yet to come to a fix.

Made a fiddle and the filtering doesnt do anything for dates. not sure why.
https://jsfiddle.net/theeedannyd/v6ww51ez/42/

Danny

**Comment by deesismyemail** [comment]
Anyone have luck with the date sort thing so far? I am still having trouble, I couldnt get the grid to load using the  link from @theeedannyd  so I created another link with the exact same code except I updated the jquery version and also had to embed it using script tags instead of adding the url to the resources.
https://jsfiddle.net/deesismyname/70rgwt8h/8/

I am guessing this is an old thread but giving it a shot in case anyone found a fix.
